### DISCUSSION TITLE
Is CollectionItemInstance's ID set correctly?

### DISCUSSION THREAD

**Discussion by rosyth** [joalla/discogs_client#114] [General] [answered]
If I have a folder instance with releases in it, I can obtain a list of the releases,  but a little mis-programming on my part caused my folder to get a few of the same releases more than once. 
So, in my investigation how to correct this with 'discogs_client' api I noticed that a collection instance ID is not the same as the release, 
However, 
```
class CollectionItemInstance(PrimaryAPIObject):
    def __repr__(self):
        return '<CollectionItemInstance {0!r} {1!r}>'.format(self.id, self.release.title)

```
will return the Release ID and not the CollectionItemInstance ID,  because the ID property of CollectionItemInstance is set as the Release ID.
```
So for example...
In [320]: instance
Out[320]: <CollectionItemInstance 2046318 'Athens'>
In [322]: instance.id
Out[322]: 2046318
In [323]: instance.instance_id
Out[323]: 1066942043
In [324]: discogs_client_user.release(2046318)
Out[324]: <Release 2046318 'Athens'>

```
From a consistency point of view,  I would expect an Instance ID to be unique per instance,  and here that is not the case.
I would expect to be able to fetch the release ID from the instance ID.

I'm just marking this as a suggestion, since the current behaviour doesn't make using CollectionItemInstances that much more difficult, and I don't know what side effects may happen if it is 'corrected'

**Comment by JOJ0** [comment, marked as answer]
I get your point and I do agree that this could be more correct but as always in IT: It's historically. The .id always was there, but only recently we've made the instance_id attr availabe.

The sideeffects would be on the social side: It would break a lot of existing programs using discogs_client.... Are we sure this is worth the hassle?

**Comment by JOJ0** [comment]
@AnssiAhola added instance_id while he implemented some new methods around collection handling (delete) I think. Too lazy to find the PR right now ;-)) Might update this answer later :-)

```
(joalla-3.9)    ~/git/discogs_client   master  git blame discogs_client/models.py | grep instance_id
8e369216 discogs_client/models.py (Anssi Ahola       2022-03-08 20:30:56 +0200 682)     instance_id = SimpleField()
2fa6dfc0 discogs_client/models.py (Anssi Ahola       2022-03-17 16:19:38 +0200 729)         instance_url = self.fetch('resource_url') + '/releases/{0}/instances/{1}'.format(instance.id, instance.instance_id)
(joalla-3.9)    ~/git/discogs_client   master  
```

And as said: I'm not sure if we do any good to our library users and existing programs if we change it.... It would mean a lot of hassle. Probably should be done by first deprecating it and preparing users for it. And to be honest I don't think we currently have the resources to handle that. 

We'll discuss it in the team anyway and I would like to thank you for being aware and pointing it out. Appreciated!

**Comment by rosyth** [comment]
I understand...
Thanks to the maintainers for adding the _instance_id_ and _remove_release_ to the api.
They helped me clean up a bit.  I'll accept the first answer as correct.
